Description: build s390, armel, mips*, iX86 debugging symbols with -gstabs
 -g exhausts memory on these arches. Thanks to Oswald Buddenhagen for the hint.
Author: Sune Vuorela <sune@debian.org>
Author: Modestas Vainius <modax@debian.org>
Author: Timo Jyrinki <timo@debian.org>
Bug-Debian: http://bugs.debian.org/528485
Bug-Debian: http://bugs.debian.org/618325
Last-Update: 2013-02-07 (add iX86)
 
--- a/Source/WebKit/WebKit1.pri.old	2012-12-18 19:04:41.000000000 +0000
+++ b/Source/WebKit/WebKit1.pri	2013-01-25 06:51:32.133738485 +0000
@@ -8,5 +8,11 @@
 SOURCE_DIR = $${ROOT_WEBKIT_DIR}/Source/WebKit
 
+equals(QT_ARCH, s390)|equals(QT_ARCH, arm)|equals(QT_ARCH, mips)|equals(QT_ARCH, i386)|equals(QT_ARCH, i686) {
+    message("s390, armel, armhf, mipsel, i386 workaround for QtWebkit: do not build with -g, but with -gstabs")
+    QMAKE_CXXFLAGS -= -g
+    QMAKE_CXXFLAGS += -gstabs
+}
+
 INCLUDEPATH += \
     $$SOURCE_DIR/qt/Api \
     $$SOURCE_DIR/qt/WebCoreSupport \
--- a/Source/WebKit2/WebKit2.pri.old	2012-12-18 19:04:26.000000000 +0000
+++ b/Source/WebKit2/WebKit2.pri	2013-01-25 06:52:07.301737594 +0000
@@ -9,6 +9,12 @@
 
 WEBKIT2_GENERATED_SOURCES_DIR = $${ROOT_BUILD_DIR}/Source/WebKit2/$${GENERATED_SOURCES_DESTDIR}
 
+equals(QT_ARCH, s390)|equals(QT_ARCH, arm)|equals(QT_ARCH, mips)|equals(QT_ARCH, i386)|equals(QT_ARCH, i686) {
+    message("s390, armel, armhf, mipsel, i686 workaround for QtWebkit: do not build with -g, but with -gstabs")
+    QMAKE_CXXFLAGS -= -g
+    QMAKE_CXXFLAGS += -gstabs
+}
+
 INCLUDEPATH += \
     $$SOURCE_DIR \
     $$SOURCE_DIR/Platform \
